islamic_prayer_times can loop forever

<!-- READ THIS FIRST:
  - If you need additional help with this template, please refer to https://www.home-assistant.io/help/reporting_issues/
  - Make sure you are running the latest version of Home Assistant before reporting an issue: https://github.com/home-assistant/core/releases
  - Do not report issues for integrations if you are using custom components or integrations.
  - Provide as many details as possible. Paste logs, configuration samples and code into the backticks.
  DO NOT DELETE ANY TEXT from this template! Otherwise, your issue may be closed without comment.
-->
## The problem
<!-- 
  Describe the issue you are experiencing here to communicate to the
  maintainers. Tell us what you were trying to do and what happened.
-->

CI Run: https://github.com/home-assistant/core/runs/896827100

`2020-07-22T02:32:03.9315802Z FAILED tests/components/islamic_prayer_times/test_init.py::test_unload_entry[pyloop]`


```
2020-07-22T02:32:02.3119783Z INFO:homeassistant.setup:Setting up islamic_prayer_times
2020-07-22T02:32:02.3119935Z INFO:homeassistant.setup:Setup of domain islamic_prayer_times took 0.0 seconds
2020-07-22T02:32:02.3120111Z DEBUG:homeassistant.components.islamic_prayer_times:Scheduling next update for Islamic prayer times
2020-07-22T02:32:02.3120300Z DEBUG:homeassistant.components.islamic_prayer_times:Midnight is after day the changes so schedule update for after Midnight the next day
2020-07-22T02:32:02.3120742Z INFO:homeassistant.components.islamic_prayer_times:Next update scheduled for: 2020-07-22 00:46:00
2020-07-22T02:32:02.3120922Z DEBUG:homeassistant.components.islamic_prayer_times:New prayer times retrieved. Updating sensors
2020-07-22T02:32:02.3121095Z DEBUG:homeassistant.core:Bus:Handling <Event component_loaded[L]: component=islamic_prayer_times>
2020-07-22T02:32:02.3121257Z INFO:homeassistant.loader:Loaded sensor from homeassistant.components.sensor
2020-07-22T02:32:02.3121415Z INFO:homeassistant.setup:Setting up sensor
2020-07-22T02:32:02.3121573Z DEBUG:homeassistant.components.islamic_prayer_times:Scheduling next update for Islamic prayer times
2020-07-22T02:32:02.3121760Z DEBUG:homeassistant.components.islamic_prayer_times:Midnight is after day the changes so schedule update for after Midnight the next day
2020-07-22T02:32:02.3122151Z INFO:homeassistant.components.islamic_prayer_times:Next update scheduled for: 2020-07-22 00:46:00
2020-07-22T02:32:02.3122330Z DEBUG:homeassistant.components.islamic_prayer_times:New prayer times retrieved. Updating sensors
2020-07-22T02:32:02.3122489Z INFO:homeassistant.setup:Setup of domain sensor took 0.0 seconds
2020-07-22T02:32:02.3122651Z DEBUG:homeassistant.core:Bus:Handling <Event component_loaded[L]: component=sensor>
2020-07-22T02:32:02.3122824Z DEBUG:homeassistant.components.islamic_prayer_times:Scheduling next update for Islamic prayer times
2020-07-22T02:32:02.3123009Z DEBUG:homeassistant.components.islamic_prayer_times:Midnight is after day the changes so schedule update for after Midnight the next day
2020-07-22T02:32:02.3123390Z INFO:homeassistant.components.islamic_prayer_times:Next update scheduled for: 2020-07-22 00:46:00
2020-07-22T02:32:02.3123645Z DEBUG:homeassistant.components.islamic_prayer_times:New prayer times retrieved. Updating sensors
2020-07-22T02:32:02.3123805Z INFO:homeassistant.components.sensor:Setting up sensor.islamic_prayer_times
2020-07-22T02:32:02.3123976Z DEBUG:homeassistant.components.islamic_prayer_times:Scheduling next update for Islamic prayer times
2020-07-22T02:32:02.3124160Z DEBUG:homeassistant.components.islamic_prayer_times:Midnight is after day the changes so schedule update for after Midnight the next day
2020-07-22T02:32:02.3124546Z INFO:homeassistant.components.islamic_prayer_times:Next update scheduled for: 2020-07-22 00:46:00
2020-07-22T02:32:02.3124716Z DEBUG:homeassistant.components.islamic_prayer_times:New prayer times retrieved. Updating sensors
2020-07-22T02:32:02.3124884Z ERROR:homeassistant.components.sensor:Error while setting up islamic_prayer_times platform for sensor
2020-07-22T02:32:02.3125036Z Traceback (most recent call last):
2020-07-22T02:32:02.3125259Z   File "/__w/core/core/homeassistant/helpers/entity_platform.py", line 180, in _async_setup_platform
2020-07-22T02:32:02.3125440Z     await asyncio.wait_for(asyncio.shield(task), SLOW_SETUP_MAX_WAIT)
2020-07-22T02:32:02.3125636Z   File "/usr/local/lib/python3.7/asyncio/tasks.py", line 416, in wait_for
2020-07-22T02:32:02.3125794Z     return fut.result()
2020-07-22T02:32:02.3126002Z   File "/__w/core/core/homeassistant/components/islamic_prayer_times/sensor.py", line 17, in async_setup_entry
2020-07-22T02:32:02.3126168Z     client = hass.data[DOMAIN]
2020-07-22T02:32:02.3126449Z KeyError: 'islamic_prayer_times'
2020-07-22T02:32:02.3126606Z DEBUG:homeassistant.components.islamic_prayer_times:Scheduling next update for Islamic prayer times
```

```

2020-07-22T02:31:59.9252365Z INFO     homeassistant.components.islamic_prayer_times:__init__.py:143 Next update scheduled for: 2020-07-22 00:56:00
2020-07-22T02:31:59.9252582Z DEBUG    homeassistant.components.islamic_prayer_times:__init__.py:168 New prayer times retrieved. Updating sensors
2020-07-22T02:31:59.9252791Z DEBUG    urllib3.connectionpool:connectionpool.py:230 Starting new HTTP connection (1): api.aladhan.com:80
2020-07-22T02:31:59.9253328Z DEBUG    urllib3.connectionpool:connectionpool.py:442 http://api.aladhan.com:80 "GET /timings/1595289600?latitude=32.87336&longitude=-117.22743&method=2 HTTP/1.1" 200 None
2020-07-22T02:31:59.9253575Z DEBUG    homeassistant.components.islamic_prayer_times:__init__.py:126 Scheduling next update for Islamic prayer times
2020-07-22T02:31:59.9253803Z DEBUG    homeassistant.components.islamic_prayer_times:__init__.py:135 Midnight is after day the changes so schedule update for after Midnight the next day
2020-07-22T02:31:59.9254245Z INFO     homeassistant.components.islamic_prayer_times:__init__.py:143 Next update scheduled for: 2020-07-22 00:56:00
2020-07-22T02:31:59.9254458Z DEBUG    homeassistant.components.islamic_prayer_times:__init__.py:168 New prayer times retrieved. Updating sensors
2020-07-22T02:31:59.9254739Z DEBUG    urllib3.connectionpool:connectionpool.py:230 Starting new HTTP connection (1): api.aladhan.com:80
2020-07-22T02:31:59.9255268Z DEBUG    urllib3.connectionpool:connectionpool.py:442 http://api.aladhan.com:80 "GET /timings/1595289600?latitude=32.87336&longitude=-117.22743&method=2 HTTP/1.1" 200 None
2020-07-22T02:31:59.9255594Z DEBUG    homeassistant.components.islamic_prayer_times:__init__.py:126 Scheduling next update for Islamic prayer times
2020-07-22T02:31:59.9255813Z DEBUG    homeassistant.components.islamic_prayer_times:__init__.py:135 Midnight is after day the changes so schedule update for after Midnight the next day
2020-07-22T02:31:59.9256257Z INFO     homeassistant.components.islamic_prayer_times:__init__.py:143 Next update scheduled for: 2020-07-22 00:56:00
2020-07-22T02:31:59.9256472Z DEBUG    homeassistant.components.islamic_prayer_times:__init__.py:168 New prayer times retrieved. Updating sensors
2020-07-22T02:31:59.9256682Z DEBUG    urllib3.connectionpool:connectionpool.py:230 Starting new HTTP connection (1): api.aladhan.com:80
2020-07-22T02:31:59.9257198Z DEBUG    urllib3.connectionpool:connectionpool.py:442 http://api.aladhan.com:80 "GET /timings/1595289600?latitude=32.87336&longitude=-117.22743&method=2 HTTP/1.1" 200 None
```

## Environment
<!--
  Provide details about the versions you are using, which helps us to reproduce
  and find the issue quicker. Version information is found in the
  Home Assistant frontend: Developer tools -> Info.
-->

- Home Assistant Core release with the issue: 
- Last working Home Assistant Core release (if known): 
- Operating environment (OS/Container/Supervised/Core): 
- Integration causing this issue: 
- Link to integration documentation on our website: 

## Problem-relevant `configuration.yaml`
<!--
  An example configuration that caused the problem for you. Fill this out even
  if it seems unimportant to you. Please be sure to remove personal information
  like passwords, private URLs and other credentials.
-->

```yaml

```

## Traceback/Error logs
<!--
  If you come across any trace or error logs, please provide them.
-->

```txt

```

## Additional information

--- a/tests/components/islamic_prayer_times/test_init.py
+++ b/tests/components/islamic_prayer_times/test_init.py
@@ -20,7 +20,7 @@ from tests.async_mock import patch
 from tests.common import MockConfigEntry, async_fire_time_changed
 
 
-async def test_setup_with_config(hass):
+async def test_setup_with_config(hass, legacy_patchable_time):
     """Test that we import the config and setup the client."""
     config = {
         islamic_prayer_times.DOMAIN: {islamic_prayer_times.CONF_CALC_METHOD: "isna"}
@@ -36,7 +36,7 @@ async def test_setup_with_config(hass):
         await hass.async_block_till_done()
 
 
-async def test_successful_config_entry(hass):
+async def test_successful_config_entry(hass, legacy_patchable_time):
     """Test that Islamic Prayer Times is configured successfully."""
 
     entry = MockConfigEntry(domain=islamic_prayer_times.DOMAIN, data={},)
@@ -55,7 +55,7 @@ async def test_successful_config_entry(hass):
         }
 
 
-async def test_setup_failed(hass):
+async def test_setup_failed(hass, legacy_patchable_time):
     """Test Islamic Prayer Times failed due to an error."""
 
     entry = MockConfigEntry(domain=islamic_prayer_times.DOMAIN, data={},)
@@ -71,7 +71,7 @@ async def test_setup_failed(hass):
         assert entry.state == config_entries.ENTRY_STATE_SETUP_RETRY
 
 
-async def test_unload_entry(hass):
+async def test_unload_entry(hass, legacy_patchable_time):
     """Test removing Islamic Prayer Times."""
     entry = MockConfigEntry(domain=islamic_prayer_times.DOMAIN, data={},)
     entry.add_to_hass(hass)

--- a/tests/components/islamic_prayer_times/test_init.py
+++ b/tests/components/islamic_prayer_times/test_init.py
@@ -33,6 +33,7 @@ async def test_setup_with_config(hass):
             await async_setup_component(hass, islamic_prayer_times.DOMAIN, config)
             is True
         )
+        await hass.async_block_till_done()
 
 
 async def test_successful_config_entry(hass):
@@ -46,6 +47,7 @@ async def test_successful_config_entry(hass):
         return_value=PRAYER_TIMES,
     ):
         await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
 
         assert entry.state == config_entries.ENTRY_STATE_LOADED
         assert entry.options == {
@@ -65,6 +67,7 @@ async def test_setup_failed(hass):
         side_effect=InvalidResponseError(),
     ):
         await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
         assert entry.state == config_entries.ENTRY_STATE_SETUP_RETRY
 
 
@@ -95,6 +98,7 @@ async def test_islamic_prayer_times_timestamp_format(hass, legacy_patchable_time
         return_value=PRAYER_TIMES,
     ), patch("homeassistant.util.dt.now", return_value=NOW):
         await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
 
         assert (
             hass.data[islamic_prayer_times.DOMAIN].prayer_times_info

